### PR TITLE
fix: validate staking amount while filling the form

### PR DESCRIPTION
**Pull request by jrwbabylonlab**

### CODE REVIEW

File: src/app/components/Staking/Form/StakingAmount.tsx [hunk 1]
```diff
@@ -51,17 +53,10 @@ export const StakingAmount: React.FC<StakingAmountProps> = ({
...
-    if (value === "") {
-      onStakingAmountSatChange(0);
-      setError(generalErrorMessage);
-      return;
-    }
-
+  useEffect(() => {
+    if (btcWalletBalanceSat === undefined) return;
     const numValue = parseFloat(value);
+    if (!numValue) return;
```
Comment (vitsalis) [inline src/app/components/Staking/Form/StakingAmount.tsx:59]: previously we were changing the staking amount and setting an error in this condition. Why not now?

File: src/app/components/Staking/Form/StakingAmount.tsx [hunk 2]
```diff
@@ -103,6 +98,24 @@ export const StakingAmount: React.FC<StakingAmountProps> = ({
...
       setValue(maxDecimals(satoshiToBtc(satoshis), 8).toString());
     }
+  }, [
+    btcWalletBalanceSat,
+    minStakingAmountSat,
+    maxStakingAmountSat,
+    value,
+    onStakingAmountSatChange,
+    coinName,
+    blured,
```
Comment (vitsalis) [inline src/app/components/Staking/Form/StakingAmount.tsx:108]: `blured` is not being used

**Review by vitsalis — COMMENTED** [review]
Some early comments. Will wait for @gbarkhatov and @jeremy-babylonlabs to test first as I'm not well versed in `useEffect` vs. `blur`

File: src/app/components/Staking/Staking.tsx [hunk 3]
```diff
@@ -366,10 +366,15 @@ export const Staking: React.FC<StakingProps> = ({
         );
         return stakingFeeSat;
       } catch (error: Error | any) {
+        let errorMsg = error?.message;
+        if (error?.message.includes("Insufficient funds")) {
```
Comment (gbarkhatov) [inline src/app/components/Staking/Staking.tsx:370]: Lets add a code that we enhance this error message so it's easier to navigate later on

**Review by gbarkhatov — APPROVED** [review]
Achieves the UX given the constraints we have, @jeremy-babylonlabs please also play around

**Review by jeremy-babylonlabs — APPROVED** [review]

**Review by vitsalis — APPROVED** [review]

**Pull request merged**